src/transformations.jl

```diff
@@ -223,17 +303,22 @@ julia> StatsBase.transform(dt, X)
  0.0  0.5  1.0
 ```
 """
-function fit(::Type{UnitRangeTransform}, X::AbstractArray{<:Real,2}; unit::Bool=true)
-    d, n = size(X)
-
-    tmin = X[:, 1]
-    tmax = X[:, 1]
-    for j = 2:n
-        @inbounds for i = 1:d
-            if X[i, j] < tmin[i]
-                tmin[i] = X[i, j]
-            elseif X[i, j] > tmax[i]
-                tmax[i] = X[i, j]
+function fit(::Type{UnitRangeTransform}, X::AbstractMatrix{<:Real};
+             dims::Union{Integer,Nothing}=nothing, unit::Bool=true)
+    if dims == 1
+        X_ = X
+    elseif dims == 2
+        X_ = X'
```

Comment:
This is going to create a type instability which could hurt performance. Better put the code below in a separate function which would take `X_` to make sure the loop is type stable.

Reply:
Yes.